src/libraries/System.Text.Json/src/System/Text/Json/Node/JsonNode.cs

```diff
@@ -0,0 +1,248 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Text.Json.Node
+{
+    /// <summary>
+    ///   The base class that represents a single node within a mutable JSON document.
+    /// </summary>
+    public abstract partial class JsonNode
+    {
+        internal const DynamicallyAccessedMemberTypes MembersAccessedOnRead =
+            DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.PublicProperties | DynamicallyAccessedMemberTypes.PublicFields;
+
+        private JsonNode? _parent;
+        private JsonNodeOptions? _options;
+
+        /// <summary>
+        ///   Options to control the behavior.
+        /// </summary>
+        public JsonNodeOptions? Options
+        {
+            get
+            {
+                if (!_options.HasValue && Parent != null)
+                {
+                    return Parent.Options;
```

Comment:
This is a design decision around how "dynamic" the options are and whether the options should change if a child node is re-parented.

Today I think it makes sense to assign it on first access per your comment (to avoid dynamic parent lookup in the future) since today case-insensitivity is the only option and that is used when creating the internal dictionary for `JsonObject` and that can't be undone reasonably when re-parenting.